core/trino-main/src/main/java/io/trino/operator/output/SlicePositionsAppender.java

```diff
@@ -131,8 +136,18 @@ public void appendRle(RunLengthEncodedBlock block)
             updateSize(rlePositionCount, 0);
         }
         else {
+            if (block.getValue() instanceof VariableWidthBlock) {
```

Comment:
I don't think you have to specialize here. You can just use block generic `io.trino.spi.block.Block#getSlice`